server/svix-server/src/worker.rs

```diff
@@ -291,6 +282,32 @@ fn bytes_to_string(bytes: bytes::Bytes) -> String {
     }
 }
 
+/// Manages preparation and execution of a QueueTask type
+async fn process_task(
+    cfg: Configuration,
+    db: &DatabaseConnection,
+    cache: Cache,
+    queue_tx: &TaskQueueProducer,
+    queue_task: QueueTask,
+) -> Result<Vec<()>> {
+    // get the message row
+    let msg = message::Entity::find_by_id(queue_task.clone().get_message_id())
+        .one(db)
+        .await
+        .unwrap()
+        .unwrap();
+
+    // get individual MessageTasks per endpoint and collect their executions as futures
+    let futures: Vec<_> = queue_task
+        .get_message_tasks()
+        .into_iter()
+        .map(|msg_task| dispatch(cfg.clone(), db, cache.clone(), queue_tx, msg_task, &msg))
+        .collect();
+
+    // execute the MessageTasks concurrently; any single failure will bubble up to the invoking function
+    future::try_join_all(futures).await
```

Comment:
Can we change this to `join_all` to avoid unnecessarily cancelling other tasks?

Reply:
Been thinking about this on my flight. If we do this, then I need to manually check each result in the vector for error types -- this functionality is not provided.

I'm fine with that.. but the more I think about it, the more I'm not sure how much sense that makes? Like, the errors we can anticipate (non-200 responses, timeouts, etc), we've optimized for and they get requeued appropriately. Errors outside of what's anticipated, I think we should just fail the whole batch atomically (or as close to atomically as we can get). If this run crashes, we should nack without regard for attempting the remaining endpoints. They're all going to get rerun anyways, because this situation is going to result in a nack regardless.

Idk, that's my take. Curious to hear @svix-daniel's and @tasn's thoughts on the matter.

Reply:
I'm with @svix-james here. I think we shouldn't cancel other tasks, but instead deal with them separately. It's not very hard to do so, and it's very useful.

This is without reviewing the rest of the PR. Just based on this small part and the discussion here.